how-it-works/1_protocol/02_consensus.subpage.md

```diff
@@ -5,14 +5,118 @@ shareImage: /img/how-it-works/consensus.600.jpg
+Block finality on the IC is cryptographic and thus can have very short latency.
+
+Consensus proceeds in discrete rounds.
+At a high level, a consensus round has the following three phases:
+* *Block making:* Creating and proposing a block containing a list of messages to be executed. In every round at least one node proposes a block, ideally exactly one node, but, if necessary, multiple nodes.
+Block proposals are broadcast by the proposer to all nodes in the subnet using P2P.
+* *Notarization:* Attesting to the validity of a block. For a block to become the block of a given round, it needs to be notarized by at least two thirds of the nodes.
+A node notarizing a block means that it considers the block to be well formed and valid and a potential candidate for the block of this round.
+* *Finalization:* Cryptographically determining the finalized block of a round. As multiple blocks may get notarized in a round, it is not immediately clear, which of those should be the block chosen for the round.
+The finalization protocol helps resolve this.
+Whenever there is exactly one fully-notarized block in a given round, the finalization protocol cryptographically ensures that every node can observe this block to be the block chosen for the round.
+If there are multiple, there is no finalization in this round, but one of the blocks becomes finalized through a future round that has only a single block fully notarized block.
+
+Let us next look at the different phases of a consensus round in more detail.
+
+## Block Making and Block Maker Selection
+
+In every round, a subnet needs to generate a *consensus block*, or *block*.
+A block contains a list of ingress messages and xnet messages.
+Ingress messages that are submitted from outside the IC to nodes of the subet are broadcast via the peer-to-peer layer to all nodes of the subnet.
+A *block maker*, i.e., a node that is supposed to create the block for the current round, is determined using a cryptographic mechanism called *random beacon* explained below.
+The so-determined block maker assembles a block from (a subset of) the ingress messages submitted directly to the node or received from other nodes through broadcast and xnet messages sent to this subnet from other subnets.
+Xnet messages are pulled from the respective sending subnets by the block maker.
+A block, once assembled, is then broadcast to all nodes of the subnet using P2P as a block proposal for the current round.
+
+If the network is unreliable or the chosen block maker corrupted or other things are going wrong with producing, broadcasting, and notarizing the block, the first proposed block may not get notarized within a reasonable timeframe or not at all.
+Thus, other nodes also need to get the opportunity to act as block makers and propose a block, so that eventually a block for the round will be successfully proposed and notarized, even under adversarial conditions, as more and more nodes attempt to propose a block.
```

Comment:
Oddly, I don't see any mention of the requirement that a proposed block extends a (fully?) notarized block.